pallets/stellar-bridge/src/lib.rs

```diff
@@ -309,6 +318,11 @@ pub mod pallet {
             }
         }
 
+        fn is_escrow(public_key_bin: [u8; 32]) -> bool {
+            let public_key = stellar::PublicKey::from_binary(public_key_bin).to_encoding();
+            return public_key.eq(&T::GatewayEscrowSecretKey::get().get_encoded_public());
```

Comment:
You can just compare the binary public key instead of the encoded public keys. Slightly more efficient.

Reply:
Done